x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/RestartIndexFollowingIT.java

```diff
@@ -87,16 +92,20 @@ public void testFollowIndex() throws Exception {
         ensureLeaderGreen("index1");
 
         final long thirdBatchNumDocs = randomIntBetween(10, 200);
+        logger.info("Indexing [{}] docs as third batch", thirdBatchNumDocs);
         for (int i = 0; i < thirdBatchNumDocs; i++) {
             leaderClient().prepareIndex("index1").setSource("{}", XContentType.JSON).get();
         }
 
-        assertBusy(
-            () -> assertThat(
-                followerClient().prepareSearch("index2").get().getHits().getTotalHits().value,
-                equalTo(firstBatchNumDocs + secondBatchNumDocs + thirdBatchNumDocs)
-            )
-        );
+        var totalDocs = firstBatchNumDocs + secondBatchNumDocs + thirdBatchNumDocs;
+        assertBusy(() -> {
+            if (hasFollowTaskWithFatalException("index2")) {
```

Comment:
I think we don't need this in the `assertBusy()`, we should only have to reconnect at most once after `fullRestart()` has returned.

Reply:
Ok, I pushed 1a481fa95dff2b0807842952d9e55092b3bfe311, let me know if that's what you have in mind.